prov/sockets/src/sock_fabric.c

```diff
@@ -580,19 +581,29 @@ SOCKETS_INI
 {
 	char *value;
 
-	if ((value = getenv("SOCK_PE_WAITTIME"))) {
+	if (fi_var_register(&sock_prov, "pe_waittime",
+				"How many miliseconds to spin while waiting for progress"))
+		SOCK_LOG_INFO("Failed to register pe_waittime variable\n");
+	if (fi_var_get(&sock_prov, "pe_waittime", &value))
+		SOCK_LOG_INFO("Failed to read pe_waittime variable\n");
```

Comment:
Fair enough -- there's actually a subtle distinction between the FI configuration variable name and how it is exposed.  Currently, FI config variables are exposed in the environment via the form $FI_(upper_case(provider_name))_(upper_case(variable_name)).  E.g.:

```
shell$ export FI_SOCKETS_PE_WAITTIME=10
```

But I would expect that if we supported config files someday, the variable name would be exposed via "(prover_name)_(variable_name)", e.g., perhaps:

```ini
sockets_pe_waittime = 10
```

(and perhaps the name in the config file would be case insensitive...?)

So there's two things different:

1. The `FI_` prefix in the env var case
1. The upper / lower case issues

I feel strongly that we need the first one (`FI_` prefix), but I could be talked out of the second (upper/lower case transmogrification).